src/overlays/actors/ovl_En_Ma_Yto/z_en_ma_yto.c

```diff
@@ -21,126 +83,1425 @@ const ActorInit En_Ma_Yto_InitVars = {
+            EnMaYto_SetFaceExpression(this, 0, 1);
+            func_801518B0(globalCtx, 0x33D0, &this->actor);
+            this->textId = 0x33D0;
+            // Attempted Cremia Cart Ride
+            gSaveContext.weekEventReg[0xE] |= 0x1;
+            this->unk_310 = 3;
+            func_80151BB4(globalCtx, 6);
+            EnMaYto_SetupPostMilkRunWaitDialogueEnd(this);
+        }
+    } else {
+        func_800B85E0(&this->actor, globalCtx, 200.0f, -1);
+    }
+}
+
+void EnMaYto_SetupBeginWarmFuzzyFeelingCs(EnMaYto* this) {
+    this->actionFunc = EnMaYto_BeginWarmFuzzyFeelingCs;
+}
+
+void EnMaYto_BeginWarmFuzzyFeelingCs(EnMaYto* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCanPlayNext(this->actor.cutscene) != 0) {
+        ActorCutscene_Start(this->actor.cutscene, &this->actor);
+        EnMaYto_SetupWarmFuzzyFeelingCs(this);
+    } else {
+        if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+            ActorCutscene_Stop(0x7C);
+        }
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+    }
+}
+
+void EnMaYto_SetupWarmFuzzyFeelingCs(EnMaYto* this) {
+    EnMaYto_SetFaceExpression(this, 0, 1);
+    this->actionFunc = EnMaYto_WarmFuzzyFeelingCs;
+}
+
+static u16 D_80B915F0 = 99;
+
+void EnMaYto_WarmFuzzyFeelingCs(EnMaYto* this, GlobalContext* globalCtx) {
+    if (func_800EE29C(globalCtx, 0x22C) != 0) {
+        u32 csActionIndex = func_800EE200(globalCtx, 0x22C);
+
+        if (globalCtx->csCtx.frames == globalCtx->csCtx.npcActions[csActionIndex]->startFrame) {
+            u16 action = globalCtx->csCtx.npcActions[csActionIndex]->unk0;
+
+            if (1) {}
+
+            if (action != D_80B915F0) {
+                D_80B915F0 = action;
+                switch (action) {
+                    case 1:
+                        EnMaYto_ChangeAnim(this, 0);
+                        break;
+
+                    case 2:
+                        // Attempted Cremia Cart Ride
+                        gSaveContext.weekEventReg[0xE] |= 1;
+                        EnMaYto_ChangeAnim(this, 18);
+                        break;
+
+                    case 3:
+                        EnMaYto_ChangeAnim(this, 22);
+                        break;
+                }
+            }
+        }
+
+        func_800EDF24(&this->actor, globalCtx, csActionIndex);
+        if (D_80B915F0 == 2 && this->skelAnime.animCurrentSeg == &D_06001FD0 &&
+            func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount)) {
+            EnMaYto_ChangeAnim(this, 20);
+        }
+    } else {
+        D_80B915F0 = 99;
+    }
+}
+
+void EnMaYto_SetupPostMilkRunWaitDialogueEnd(EnMaYto* this) {
+    this->actionFunc = EnMaYto_PostMilkRunWaitDialogueEnd;
+}
+
+void EnMaYto_PostMilkRunWaitDialogueEnd(EnMaYto* this, GlobalContext* globalCtx) {
+    if (func_80152498(&globalCtx->msgCtx) == 6 || func_80152498(&globalCtx->msgCtx) == 5) {
+        if (func_80147624(globalCtx) != 0 && func_80152498(&globalCtx->msgCtx) == 5) {
```

Comment:
You could probably combine these ifs into one

Reply:
Yeah probably, but I prefered to kept them separated, I think is a bit more readable this way.